Set up configuration and initialize connection to Kentix endpoints

zdevaty opened the pull request.

Commit:
feat(SB-05320): communicate with first Kentix endpoint

Commit:
feat(SB-05320): respect configurations when querying data

Commit:
fix(SB-05320): remove trailing commas in JSON

Commit:
fix(SB-05320): update eliona dependencies

Commit:
feat(SB-05230): make the device info fetching general

Also, fix a data race.

Commit:
fix(SB-05320): allow gracefully stopping

Commit:
fix(SB-05320): generalize fetching data

Commit:
feat(SB-05320): Implement fetching data from MultiSensor

andreas-goebel commented on kentix/kentix.go:
```diff
@@ -59,31 +65,83 @@ func GetDeviceInfo(conf apiserver.Configuration) (*DeviceInfo, error) {
 func fetchData(url string, dest interface{}) error {
 	client := &http.Client{}
 
 	req, err := http.NewRequest("GET", url, nil)
```
For an API access, please use our methods from go-utils/http. At the moment there is no method for authorization Basic with an API-Key, but I will create such a method. The difference between the API mock and real Kentic devices is, that the devices needs an authorization. You can see that in https://dev.azure.com/eliona/apps/_git/kentix?path=/kentix_doorlock_get_data.sh

The methods for go-utils/http can you find here: https://github.com/eliona-smart-building-assistant/go-utils/tree/main/http

andreas-goebel replied:
There is already a method to do this:
http.NewRequestWithApiKey(url, "Authorization", "Basic " + API-Key from config in Base64)

andreas-goebel replied:
Pleae note, that the  API-Key from config from config have to encoded in Base64. See: https://dev.azure.com/eliona/apps/_git/kentix?path=/kentix_doorlock_get_data.sh

andreas-goebel replied:
With the method you can read the API in 2 lines:
request := http.NewRequestWithApiKey(url, "Authorization", "Basic " + conf.apiKey)
sensorData := http.Read[SensorData](request, 10, true)

zdevaty replied:
Thanks, I implemented it in the next CL.

andreas-goebel reviewed (changes requested):
for API access, please use our methods from go-utils/http

Commit:
feat(ASB-0017): fetch doorlocks

The doorlocks list contains pagination, which is undocumented. I
implemented that as well.

Commit:
feat(SB-05320): Authenticate API requests

Commit:
fix: make log message tags more meaningful

andreas-goebel committed:
Merge pull request #4 from eliona-smart-building-assistant/ASB-0017

feat(SB-05320): Implement fetching doorlocks

andreas-goebel committed:
Merge pull request #5 from eliona-smart-building-assistant/fix-logs

fix: make log message tags more meaningful
